keras_cv/layers/preprocessing/aug_mix.py

```diff
@@ -0,0 +1,301 @@
+            A value is sampled from the provided range.  If a float is passed, the
+            range is interpreted as `(0, severity)`. This value represents the
+            level of strength of augmentations and is in the range [0, 1].
+            Defaults to 0.3.
+        width: an integer representing the number of different chains to
+            be mixed. Defaults to 3.
+        depth: an integer representing the number of transformations in
+            the chains. A negative value enables stochastic depth uniformly
+            in [1,3]. Defaults to -1.
+        alpha: a float value used as the probability coefficients for the
+            Beta and Dirichlet distributions. Defaults to 1.0.
+        seed: Integer. Used to create a random seed.
+
+    References:
+        [AugMix paper](https://arxiv.org/pdf/1912.02781)
+        [Official Code](https://github.com/google-research/augmix)
+        [Unoffial TF Code](https://github.com/szacho/augmix-tf)
+
+    Sample Usage:
+    ```python
+    (images, labels), _ = tf.keras.datasets.cifar10.load_data()
+    augmix = keras_cv.layers.preprocessing.mix_up.AugMix([0, 255])
+    augmented_images = augmix(images)
+    ```
+    """
+
+    def __init__(
+        self,
+        value_range,
+        severity=0.3,
+        width=3,
+        depth=-1,
+        alpha=1.0,
+        seed=None,
+        **kwargs,
+    ):
+        super().__init__(seed=seed, **kwargs)
+        self.value_range = value_range
+        self.width = width
+        self.depth = depth
+        self.alpha = alpha
+        self.seed = seed
+        self.auto_vectorize = False
+        self.severity = severity
+        self.severity_factor = preprocessing.parse_factor(
+            self.severity, min_value=0.01, param_name="severity", seed=self.seed
+        )
+
+        # initialize layers
+        self.auto_contrast = layers.AutoContrast(value_range=self.value_range)
+        self.equalize = layers.Equalization(value_range=self.value_range)
+
+    @staticmethod
+    def _sample_from_dirichlet(alpha):
+        gamma_sample = tf.random.gamma(shape=(), alpha=alpha)
+        return gamma_sample / tf.reduce_sum(gamma_sample, axis=-1, keepdims=True)
+
+    @staticmethod
+    def _sample_from_beta(alpha, beta):
+        sample_alpha = tf.random.gamma((), 1.0, beta=alpha)
+        sample_beta = tf.random.gamma((), 1.0, beta=beta)
+        return sample_alpha / (sample_alpha + sample_beta)
+
+    def _loop_on_depth(self, depth_level, image_aug):
+        op_index = self._random_generator.random_uniform(
+            shape=(), minval=0, maxval=8, dtype=tf.int32
+        )
+        image_aug = self._apply_op(image_aug, op_index)
+        depth_level += 1
+        return depth_level, image_aug
+
+    def _loop_on_width(self, image, chain_mixing_weights, curr_chain, result):
+        image_aug = tf.identity(image)
+        depth = tf.cond(
+            tf.greater(self.depth, 0),
```

Comment:
I think this is really confusing due to the check for -1.

It would be a lot simpler if you could write:

```
# in constructor parse depth and assign it to a pair of ints always
def sample_depth(self):
    return self._random_generator.random_uniform(shape=(), minval=self.depth[0], maxval=self.depth[1], dtype=tf.int32)
depth = self.sample_depth()
```